### DISCUSSION TITLE
How to read or write the properties block of molfiles using RDKit?

### DISCUSSION THREAD

**Discussion by lucas-morin** [rdkit/rdkit#5634] [Ideas] [answered]
Hello,
Do you know if RDKit allows to read or write the property block of molfiles?

For example, in the following molfile, I would like to read the abbreviation "OCH3" directly from the S-group definition.
```
from rdkit import Chem

molecule = Chem.MolFromMolBlock("""0121.cdx
  ChemDraw04300520492D

 12 13  0  0  0  0  0  0  0  0999 V2000
   -0.9920    0.7219    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.9920   -0.1031    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.2775   -0.5156    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.4369   -0.1031    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    0.4369    0.7219    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.2775    1.1344    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.7065   -0.5156    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.2216   -0.3581    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    1.7065    0.3094    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2216    0.9768    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
   -0.6348   -0.7219    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.0797   -1.1344    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  4  0      
  2  3  4  0      
  3  4  4  0      
  4  5  4  0      
  5  6  4  0      
  6  1  4  0      
  2  7  1  0      
  4  8  1  0      
  8  9  1  0      
  9 10  1  0      
 10  5  1  0      
  3 11  1  0      
 11 12  1  0      
M  STY  1   1 SUP
M  SLB  1   1   1
M  SAL   1  2  11  12
M  SBL   1  1  12
M  SMT   1 OCH3
M  SBV   1 12   -2.4599    6.1489
M  END""")
```

Here is the extracted structure:
![rdkit_molecule](https://user-images.githubusercontent.com/73230090/194914541-19cce88b-da11-4013-a16e-93e73f914e9f.png)

Ideally, the atoms corresponding to "OCH3" would be merged to a single node, having a display label "OCH3", like in the following structure:
![US07314693-20080101-C00752](https://user-images.githubusercontent.com/73230090/194914439-7e431485-1c55-4bf7-ad86-04b16f23c220.png)

Conversely, I would like to change the display label of a node and preserve this information when saving the molecule. 
```
from rdkit import Chem
molecule = Chem.MolFromSmiles("CCCCC")
molecule.GetAtomWithIdx(0).SetProp("_displayLabel", "test") 
rdmolfiles.MolToMolFile(molecule, "test.mol")
```

Are these features supported in RDKit?

**Comment by greglandrum** [comment, marked as answer]
@lucasmorin222, the `rdkit.Chem.rdAbbreviations` module has code for working with abbreviations.
There's info on this in the cookbook: https://www.rdkit.org/docs/Cookbook.html#with-abbreviations
And here is a demo with your molecule:
https://gist.github.com/greglandrum/f86587a196da572a2f8b813e429c19e7

**Reply by lucas-morin** [reply]
Thanks a lot, this is what I was looking for.
